java/code/src/com/redhat/rhn/domain/rhnpackage/Package.java

```diff
@@ -55,6 +55,8 @@ public class Package extends BaseDomainHelper {
     private String path;
     private String headerSignature;
     private String copyright;
+    private Long isPtfPackage = 0L;
```

Comment:
why do we need both flags? What is the difference between the two flags?

Reply:
There are two different flags here:

1) `is_ptf` -> only for packages which provides `ptf()` attribute. This is used for the kind of metapackage for the PTF, named like: `ptf-ABCDE-VER-REL.PTF.xxxxxx.noarch.rpm`

2) `is_part_of_ptf` -> for those packages that are parf of a PTF (the actual patched package), like `salt-3006.0-xx.PTF.xxxxx.noarch.rpm`

I see our SQL queries are looking / filtering based on those two, so that is why added the two attributes so we can improve performance for the two attributes.

Reply:
ok, got your point. When we are looking at the package capabilities we need to look for both to know if it's a PTF. 
But my question was more in the direction of whether we need to save both values separately fields or just one database field would be enough (like is_ptf, that would be true if the package have the capability `ptf()` of `ptf-package()`)

Reply:
In Uyuni I see there are currently different behaviors for PTF metapackages (the ones providing `ptf()` as capabilities) and the packages that are part of a PTF (the other ones with only `ptf-package()` capability).

Since we do things differently for each type of package, IMO we should store those two attributes separetely to gain on performance anytime we need to differentiate between "normal packages / metapackages / packages part of ptfs".

I hope this makes sense to you and I'm not missing anything! :)

Reply:
ok, makes sense to me. All god from my side :)
 If we have both we can better control and decide to which we are looking for. BUt we must be aware that we need to look at both flags when we are implementing a feature related to PTF.